build-setup/windows.md

```diff
@@ -119,33 +119,21 @@ nmake
 nmake install
 ```
 
-### Boost 1.77.0
+### Boost 1.83.0
 
-Download version 1.77.0 of Boost from <https://boostorg.jfrog.io/artifactory/main/release/1.77.0/source/boost_1_77_0.tar.bz2> . You must get this exact version, as the Machine Learning build system requires it.
+Download version 1.83.0 of Boost from <https://boostorg.jfrog.io/artifactory/main/release/1.83.0/source/boost_1_83_0.tar.bz2> . You must get this exact version, as the Machine Learning build system requires it.
 
 Assuming you chose the `.bz2` version, extract it in a Git bash shell using the GNU tar that comes with Git for Windows, e.g.:
 
 ```
 cd /c/tools
-tar jxvf /z/cpp_src/boost_1_77_0.tar.bz2
-```
-
-Edit `boost/unordered/detail/implementation.hpp` and change line 287 from:
-
-```
-    (17ul)(29ul)(37ul)(53ul)(67ul)(79ul) \
-```
-
-to:
-
-```
-    (3ul)(17ul)(29ul)(37ul)(53ul)(67ul)(79ul) \
```

Comment:
Why is this substitution no longer used? Did the Boost maintainers decide to start off with a smaller hash table in the official code? This optimisation was originally included because the anomaly detection code creates large numbers of hash maps with one item in them, and reserving 17 slots in such cases noticeably increases the memory usage of the `autodetect` process for high cardinality jobs.

Reply:
The `BOOST_UNORDERED_PRIMES` definition is no longer present after `Boost 1.79`, so there's no longer anything to substitute.

Reply:
Please can you grep the unordered code and check if another file has a similar set of primes in it. If you grep for 17 and 29 on the same line that might find it. I’m a bit worried we’ll get reports of greatly increased model memory sizes if the memory usage of a size 1 hash map dramatically increases.

Reply:
I suggest adding `(3ul)` to the beginning of that line and see if everything still works and memory usage is reduced.